lib/Dialect/Seq/SeqOps.cpp

```diff
@@ -597,6 +620,44 @@ OpFoldResult FirRegOp::fold(FoldAdaptor adaptor) {
   return IntegerAttr::get(intType, 0);
 }
 
+//===----------------------------------------------------------------------===//
+// ClockGateOp
+//===----------------------------------------------------------------------===//
+
+OpFoldResult ClockGateOp::fold(FoldAdaptor adaptor) {
+  // Forward the clock if one of the enables is always true.
+  if (isConstantOne(adaptor.getEnable()) ||
+      isConstantOne(adaptor.getTestEnable()))
```

Comment:
Does this need to check for existence of the enable first?

Reply:
That's done in `isConstantOne`. The `adaptor` will return null if the enable isn't there, and the function handles that case.